__tests__/app.test.js

```diff
@@ -144,10 +144,144 @@ describe("app", () => {
+    .get(`/api/articles/${testArticleId}/comments`)
+    .expect(200)
+    .then((response) => {
+      const {comments} = response.body;
+      expect(comments.length).not.toEqual(0);
+      comments.forEach((comment) => {
+        expect(comment).toBeInstanceOf(Object);
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+      });
+    });
+  });
+  test("200: responds with comments sorted descending based on created_at when comments for an article are found", () => {
+    const testArticleId = 1;
+    return request(app)
+    .get(`/api/articles/${testArticleId}/comments`)
+    .expect(200)
+    .then((response) => {
+      const { comments } = response.body;
+      expect(comments).toBeSortedBy('created_at', { descending: true });
+    })
+  });
+ });
+ describe("POST /api/articles/:article_id/comments", () => {
+  test("201: adding a valid comment for existing article responds with status 201", () => {
+    const testArticleId = 4;
+    const requestBody = {
+      username: 'icellusedkars',
+      body: 'Lobster pot'
+    };
+    return request(app)
+     .post(`/api/articles/${testArticleId}/comments`)
+     .send(requestBody)
+     .expect(201);
+  });
+  test("201: adding a valid comment for existing article responds with added comment", () => {
+    const testArticleId = 4;
+    const requestBody = {
+      username: 'icellusedkars',
+      body: 'Lobster pot'
+    };
+
+    return request(app)
+     .post(`/api/articles/${testArticleId}/comments`)
+     .send(requestBody)
+     .expect(201)
+     .then((response) => {
+        const comment = response.body.comment;
+        expect(comment).toHaveProperty('article_id', testArticleId);
+        expect(comment).toHaveProperty('author', requestBody.username);
+        expect(comment).toHaveProperty('body', requestBody.body);
+        expect(comment).toHaveProperty('votes', 0);
+     });
+  });
+  test("400: adding a valid comment for an article that does not exist responds with status 400", () => {
+    const testArticleId = 4000;
+    const requestBody = {
+      username: 'icellusedkars',
+      body: 'Lobster pot'
+    };
+    return request(app)
+     .post(`/api/articles/${testArticleId}/comments`)
+     .send(requestBody)
+     .expect(400);
+  });
+  test("400: adding a valid comment for a user that does not exist responds with status 400", () => {
+    const testArticleId = 4;
+    const requestBody = {
+      username: 'notrealuser',
+      body: 'Lobster pot'
+    };
+    return request(app)
+     .post(`/api/articles/${testArticleId}/comments`)
+     .send(requestBody)
+     .expect(400);
```

Comment:
Same with this one its a 404 it's a valid request but the username was not found.